### PR TITLE
Funnel step querying UI

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

- Includes redesigned funnel step sidebar, including query filter UI and chart type selection.
- Removes "Calculate" button on funnels, and auto-fetches funnel data whenever a step or filter is changed.
- Makes Reset and Save to Dashboard buttons consistent with new Insights UI visual design
- Removes blue "Add to dashboard" button on insights display config (header bar of visualization). It was previously only used on funnels.

Related PR: #4944
Depends on feature flag `4535-funnel-bar-viz`

Screen Shot 2021-06-30 at 10 16 58 AM

- [x] Adds series numbers to funnel step query.
- [x] Make dropdown and button fullwidth
- [x] Graph type toggle
- [x] Save button
- [ ] Conversion window - pending backend implementation?

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [x] Frontend/CSS is usable at 320px (iPhone SE) and decent at 360px (most phones)
- [x] Breaking changes are backwards-compatible. Ensure old/new frontend requests work with new/old backends, and vice versa.

### CODE REVIEW

**Comment by clarkus** [comment]
Some quick feedback on affordance. We would see a benefit from using a different drag indicator here. The current icon is a vertical ellipsis which most users will identify as an overflow menu or some action that shows more actions. Doubling up into two columns of repeated shapes creates a textural feel to the icon and can be more obvious for indicating drag and drop actions. 

Screen Shot 2021-06-29 at 10 17 39 AM

**Comment by samwinslow** [comment]
Converting to draft; will only hide the Calculate button for Clickhouse users as Postgres query performance might still be problematic.

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx [hunk 1]
```diff
@@ -12,10 +12,18 @@ import { TestAccountFilter } from 'scenes/insights/TestAccountFilter'
...
+import { ToggleButtonChartFilter } from './ToggleButtonChartFilter'
+import { InsightActionBar } from '../InsightActionBar'
+import { preflightLogic } from 'scenes/PreflightCheck/logic'
 
 export function FunnelTab(): JSX.Element {
     useMountedLogic(funnelCommandLogic)
     const { isStepsEmpty, filters, stepsWithCount } = useValues(funnelLogic())
+    const { featureFlags } = useValues(featureFlagLogic)
+    const { preflight } = useValues(preflightLogic)
+    const autoCalculate = featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ] && preflight?.is_clickhouse_enabled // Query performance is suboptimal on psql
```
Comment (mariusandra) [inline frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:26]: This is the second place. Could be a selector from `funnelLogic` for convenience

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 2]
```diff
@@ -202,7 +206,11 @@ export const funnelLogic = kea<funnelLogicType>({
             actions.setStepsWithCountLoading(false)
         },
         setFilters: ({ refresh }) => {
-            if (refresh) {
+            // FUNNEL_BAR_VIZ removes the Calculate button
+            // Query performance is suboptimal on psql
+            const autoCalculate =
+                values.featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ] && values.preflight?.is_clickhouse_enabled
```
Comment (mariusandra) [inline frontend/src/scenes/funnels/funnelLogic.ts:212]: This `autoCalculate` could be a selector. This way the logic doesn't need to be duplicated in two additional components.

File: frontend/src/scenes/insights/Insights.tsx [hunk 3]
```diff
@@ -393,7 +394,9 @@ function FunnelInsight(): JSX.Element {
         filters: { display },
     } = useValues(funnelLogic({}))
     const { featureFlags } = useValues(featureFlagLogic)
+    const { preflight } = useValues(preflightLogic)
     const fluidHeight = featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ] && display === FUNNEL_VIZ
+    const autoCalculate = featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ] && preflight?.is_clickhouse_enabled
```
Comment (mariusandra) [inline frontend/src/scenes/insights/Insights.tsx:399]: And a third place :)

**Review by mariusandra — APPROVED** [review]
Looks good! I have one suggestion for simplification, 👍 for the rest!

**Pull request merged**